### PR TITLE
docs: blog post for works and workplacements

### LINKED ISSUE #119: Blog: Publish a blog on decompressing works when debugging Kratix
As introduced in [the feature on reducing works size](https://github.com/syntasso/kratix/issues/226), users debugging kratix will now need to decompress the workplacement contents. We will be describing how to do this to docs, but to make it easier for someone to know when they might come across this, we want a technical blog that describes:
1. When to check the works if you get an error in kratix
2. How to decompress the bits you need to decompress
3. What the contents mean and some actions you might need to take based on that fact.

### Done when
- [x] Blog reviewed by someone in the team
- [x] Blog published to kratix.io

### PR DESCRIPTION
**Pull request by kirederik**
closes #119

### CODE REVIEW

File: blog/2024-11-20-decompressing-work/index.mdx [hunk 1]
```diff
@@ -0,0 +1,211 @@
+---
+slug: from-platform-to-destination
+title: "From Platform to Destination"
```
Comment (catmo-syntasso) [inline blog/2024-11-20-decompressing-work/index.mdx:3]: Title should be a little clearer - "How your Resources get from Promise to Destination" or "Explaining the mysterious Works and WorkPlacements..." or something

Comment (catmo-syntasso) [reply blog/2024-11-20-decompressing-work/index.mdx:3]: "One for the observers: What Works and WorkPlacements do"

File: blog/2024-11-20-decompressing-work/index.mdx [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+will use the default settings.
+
+So what happens if a Work object exceeds 1.5MB? The Configure Workflow fails at
+the `work-writer` container, and the error message isn't particularly helpful:
+
+```shell-session
+etcdserver: request is too large
+```
+
+> You may see an error message like `Request entity too large: limit is
```
Comment (catmo-syntasso) [inline blog/2024-11-20-decompressing-work/index.mdx:142]: Should this be an info callout?

File: blog/2024-11-20-decompressing-work/index.mdx [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+example, includes 3.7MB of YAML!
+
+This brings us back to the [change introduced by
+243](https://github.com/syntasso/kratix/pull/243). In this update, we introduced
+gzip compression for the Workload contents before persisting the Work into etcd.
+This significantly reduces the size of the Work object (gzip documentation
+mentions an average of [70% reduction in
+size](https://www.gnu.org/software/gzip/manual/gzip.html). For the Prometheus
+Operator, for example, the size of the Work object went from 3.7MB to about
+490KB—an 87% reduction 🎉!
```
Comment (catmo-syntasso) [inline blog/2024-11-20-decompressing-work/index.mdx:158]: Missing )

File: blog/2024-11-20-decompressing-work/index.mdx [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+:::
+
+Despite compression, large Work objects may still pose challenges. While this
+update provides temporary relief, we’ll need to revisit the structure to allow
+users to have unbounded fun with their Promises. But that’s a story for another
+day.
+
+## Conclusion
+
+In this post, we've dived into the internals of Kratix to understand how a
```
Comment (catmo-syntasso) [inline blog/2024-11-20-decompressing-work/index.mdx:201]: we've dived -> we dived

File: blog/2024-11-20-decompressing-work/index.mdx [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+import Figure03 from "./figure03.png"
+import Figure04 from "./figure04.png"
+```
+
+The Kratix Team has recently introduced a
+[change](https://github.com/syntasso/kratix/pull/243) to Kratix to reduce the
+size of the Work object. While this change is mostly internal, it's a good
+opportunity to dive into how Kratix actually works and understand how a document
+moves from the Platform Cluster to a Destination.
+
```
Comment (catmo-syntasso) [inline blog/2024-11-20-decompressing-work/index.mdx:21]: You have a good conclusion at the end, might be worth an intro as well.

```
This is a fun-packed blog where you'll learn:
- how Kratix moved documents from Platform to Destinations
- what works and workplacements are
- how to inspect works to debug your Promises
```

File: blog/2024-11-20-decompressing-work/index.mdx [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+Despite compression, large Work objects may still pose challenges. While this
+update provides temporary relief, we’ll need to revisit the structure to allow
+users to have unbounded fun with their Promises. But that’s a story for another
+day.
+
+## Conclusion
+
+In this post, we've dived into the internals of Kratix to understand how a
+document moves from the Platform Cluster to a Destination. We've seen how the
```
Comment (catmo-syntasso) [inline blog/2024-11-20-decompressing-work/index.mdx:202]: we've seen -> we saw

File: blog/2024-11-20-decompressing-work/index.mdx [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+Work and WorkPlacement objects are used to schedule and write documents to the
+State Store. We've also seen how the recent change to compress the Workload
```
Comment (catmo-syntasso) [inline blog/2024-11-20-decompressing-work/index.mdx:204]: we've also seen -> we also saw

File: blog/2024-11-20-decompressing-work/index.mdx [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+contents has helped reduce the size of the Work object.
+
+We hope this post has given you a better understanding of how Kratix works under
+the hood. If you have any questions or feedback (or want to see more blog posts
+like this) please don't hesitate to reach out to us on
+[Slack](https://kratixworkspace.slack.com/) or
+[GitHub](https://github.com/syntasso/kratix).
```
Comment (catmo-syntasso) [inline blog/2024-11-20-decompressing-work/index.mdx:211]: I noticed there is a LOT of white space at the bottom of the blog page when i spun it up locally 👀 
Screenshot 2024-11-22 at 10 16 59

**Review by catmo-syntasso — CHANGES_REQUESTED** [review]

File: blog/2024-11-20-decompressing-work/index.mdx [hunk 9]
```diff
@@ -0,0 +1,211 @@
...
+---
+
+```mdx-code-block
+import Figure01 from "./figure01.png"
+import Figure02 from "./figure02.png"
+import Figure03 from "./figure03.png"
+import Figure04 from "./figure04.png"
+```
+
+The Kratix Team has recently introduced a
```
[the remark this answers is not in the document]
Comment (kirederik) [reply blog/2024-11-20-decompressing-work/index.mdx:16]: 🤷🏽‍♂️

File: blog/2024-11-20-decompressing-work/index.mdx (hunk 1 above)
Comment (kirederik) [reply blog/2024-11-20-decompressing-work/index.mdx:3]: went with How your Resources get from Promise to Destination

File: blog/2024-11-20-decompressing-work/index.mdx (hunk 8 above)
Comment (kirederik) [reply blog/2024-11-20-decompressing-work/index.mdx:211]: i think thats you... 🤠

image

**Pull request merged**